lib/moduletools.js

```diff
@@ -30,14 +35,65 @@ const moduletools = {
     return ['verbose', 'debug', 'silly'].indexOf(logger.level) >= 0;
   },
 
-  getModules(rootFolder) {
+  getAllModulesAndInstalledDependenciesDeep(location, folderName) {
+    if (location === null || location === undefined) {
+      location = process.cwd();
+    }
+
+    if (folderName === null || folderName === undefined) {
+      folderName = this.modulesFolder;
+    }
+
+    const result = {
+      modules: [],
+      installedDependencies: [],
+    };
+
+    // get the local modules and the installed modules within the current path
+    return Promise.all([
+      ModuleInfo.loadFromFolder(location, ''),    // get infos about the current module
+      this.getModules(location, 'node_modules'),  // get infos about installed modules in the current module
+      this.getModules(location, folderName),      // get infos about local submodules of the current module
```

Comment:
You can spare the comments, if you extract this into functions and give them a proper name.